src/Adapter/Cache/CombineCompressCacheConfiguration.php

```diff
@@ -0,0 +1,230 @@
+                        'parameters' => array(
+                            '%directorypath%' => $this->getThemeCacheFolder()
+                        )
+                    );
+                }
+            }
+
+            // Manage Apache optimization
+            $apacheError = $this->manageApacheOptimization((bool) $configuration['apache_optimization']);
+
+            if (count($apacheError) > 0) {
+                $errors[] = $apacheError;
+            }
+        }
+
+        return $errors;
+    }
+
+    /**
+     * @{inheritdoc}
+     */
+    public function validateConfiguration(array $configuration)
+    {
+        $resolver = new OptionsResolver();
+        $resolver->setRequired(
+            array(
+                'smart_cache_css',
+                'smart_cache_js',
+                'apache_optimization',
+            )
+        );
+        $resolver->resolve($configuration);
+
+        return true;
+    }
+
+    /**
+     * @return string Absolute path the the current active theme
+     */
+    private function getThemeCacheFolder()
+    {
+        return $this->themePath . '/' . $this->themeName . '/cache/';
+    }
+
+    /**
+     * Creates Cache folder for the active theme
+     * @return bool
+     */
+    private function createThemeCacheFolder()
+    {
+        try {
+            $folder = $this->getThemeCacheFolder();
+            $this->filesystem->mkdir($folder, '0777');
+
+            return true;
+        } catch (IOExceptionInterface $e) {
+            return false;
+        }
+    }
+
+    /**
+     * Update Cache version of assets if needed
+     * @param array the configuration
+     */
+    private function updateCachesVersionsIfNeeded(array $configuration)
+    {
+        $cacheCSS = $configuration['smart_cache_css'];
+        $currentCacheCSS = $this->configuration->get('PS_CSS_THEME_CACHE');
+
+        $cacheJS = $configuration['smart_cache_js'];
+        $currentCacheJS = $this->configuration->get('PS_JS_THEME_CACHE');
+
+        if ($cacheCSS !== $currentCacheCSS) {
+            $cssCacheVersion = $this->configuration->get('PS_CCCCSS_VERSION') + 1;
+            $this->configuration->set('PS_CCCCSS_VERSION', $cssCacheVersion);
+            $this->configuration->set('PS_CSS_THEME_CACHE', $cacheCSS);
+        }
+
+        if ($cacheJS !== $currentCacheJS) {
+            $jsCacheVersion = $this->configuration->get('PS_CCCJS_VERSION') + 1;
+            $this->configuration->set('PS_CCCCSS_VERSION', $jsCacheVersion);
+            $this->configuration->set('PS_JS_THEME_CACHE', $cacheJS);
+        }
+    }
+
+    /**
+     * Creates .htaccess if Apache optimization feature is enabled
+     * @param bool $enabled
+     * @return array not empty in case of error
+     */
+    private function manageApacheOptimization($enabled)
+    {
+        $errors = array();
+        $isActuallyEnabled = (bool) $this->configuration->get('PS_HTACCESS_CACHE_CONTROL');
```

Comment:
I guess you meant `Currently` instead of `Actually`?

Reply:
yes :)